rg3d-core/src/pool.rs

```diff
@@ -811,13 +811,25 @@ impl<T> Pool<T> {
 
     #[inline]
     #[must_use]
-    pub fn handle_from_index(&self, n: usize) -> Handle<T> {
+    pub fn at_pair(&self, n: usize) -> Option<(Handle<T>, &T)> {
+        self.handle_from_index(n).zip(self.at(n))
```

Comment:
I think it does double validity checks:

1) `handle_from_index` checks if index and generation are both valid
2) `at` checks if index is valid

I'm not sure if compiler smart enough to eliminate such duplicated checks, so maybe it is worth to combine internals of `handle_from_index` + `at` to make more efficient version? I'm concerned about this method because it is used for iterators and **may** slowdown iterators. Possible replacement:

```rust
 pub fn at_pair(&self, n: usize) -> Option<(Handle<T>, &T)> {
	if let Some(record) = self.records.get(n) {
		if record.generation != INVALID_GENERATION {
			if let Some(payload) = record.payload.as_ref() {
				return Some((Handle::new(n as u32, record.generation), payload));
			}
		}
	}
	None
}
```